compiler/noirc_evaluator/src/ssa/ir/cfg.rs

```diff
@@ -148,6 +148,48 @@ impl ControlFlowGraph {
     pub(crate) fn compute_entry_blocks(&self) -> Vec<BasicBlockId> {
         self.data.keys().filter(|&&block| self.predecessors(block).len() == 0).copied().collect()
     }
+
+    /// Computes the reverse graph of the extended CFG.
+    /// The extended CFG is the CFG with an additional unique exit node (if there is none)
+    /// such that there is a path from every block to the exit node.
+    pub(crate) fn extended_reverse(func: &mut Function) -> Self {
+        let mut cfg = Self::with_function(func);
+        // Exit blocks are the ones having no successor
+        let exit_nodes: Vec<BasicBlockId> =
+            cfg.data.keys().filter(|&&block| cfg.successors(block).len() == 0).copied().collect();
+        // Traverse the reverse CFG from the exit blocks
+        let reverse = cfg.reverse();
+        let post_order = crate::ssa::ir::post_order::PostOrder::with_cfg(&reverse);
+        // Extract blocks that are not reachable from the exit blocks
+        let dead_blocks: Vec<BasicBlockId> = cfg
+            .data
+            .keys()
+            .filter(|&block| !post_order.as_slice().contains(block))
```

Comment:
Perhaps you could collect into a `HashSet` before, to avoid having `O(n)` lookups?